polygeist/lib/Dialect/Polygeist/Transforms/SYCLRaiseHost.cpp

```diff
@@ -107,6 +210,87 @@ struct RaiseKernelName : public OpRewritePattern<LLVM::AddressOfOp> {
                                        : std::nullopt;
   }
 };
+
+/// Raise constructs assigning a kernel name to a handler to
+/// `sycl.host.handler.set_kernel`.
+///
+/// This pattern acts on `FunctionOpInterface` instances as it will not be
+/// removing the operations assigning the kernel name, but creating additional
+/// ones to mark the construct.
+struct RaiseSetKernel : public OpInterfaceRewritePattern<FunctionOpInterface> {
```

Comment:
According to the LLVM style guide, `struct` should only be used if *all* members are public: 
> struct should be used when all members are declared public